### PR TITLE
refactor: convert samples test from ava to mocha

### PR DESCRIPTION
**Pull request by nareshqlogic**
Fixes convert all sample tests from ava to mocha #2865(googleapis/google-cloud-node#2865) (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
@DominicKramer this is same as the other, we're trying to consistently use mocha for all the tests.

File: samples/system-test/express.test.js [hunk 1]
```diff
@@ -16,18 +16,18 @@
...
 const logging = new Logging();
 const got = require('got');
 
-test.before(tools.checkCredentials);
+before(tools.checkCredentials);
 
-test.serial(`should write using bunyan`, async t => {
-  t.plan(4);
+it(`should write using bunyan`, async () => {
+  //t.plan(4);
```
Comment (JustinBeckwith) [inline samples/system-test/express.test.js]: just remove this

**Comment by DominicKramer** [comment]
@JustinBeckwith ok sounds good :+1:

File: samples/system-test/express.test.js (hunk 1 above)
Comment (nareshqlogic) [reply samples/system-test/express.test.js]: Okay I will remove.

**Comment by nareshqlogic** [comment]
Hi @JustinBeckwith ,
We are getting the value of constant APP_LOG_SUFFIX is undefined as there is no property APP_LOG_SUFFIX in lb.express object. So let me know can I remove two lines 
**const lb = require('@google-cloud/logging-bunyan');
const {APP_LOG_SUFFIX} = lb.express;** 
and also change the line no 48 
**from  const log = logging.log(`samples_express_${APP_LOG_SUFFIX}`);
to const log = logging.log(`samples_express`);**

**Comment by JustinBeckwith** [comment]
👋 Thanks for the heads up.  I think someone on @googleapis/node-team may be better suited to answer this though :)

**Comment by ofrobots** [comment]
@JustinBeckwith the export `APP_LOG_SUFFIX` was added in https://github.com/googleapis/nodejs-logging-bunyan/commit/5ab87244ec45742b1603ca7347b696c9838a9876. This has not been released yet.

I guess what this implies is that we cannot update the samples at the same time as adding a feature to the library – because the samples can only test released versions. IMHO, it would be unnecessary hardship for features to be split so that the sample change happens separately after the feature gets released. Either we have to figure out a better way to test samples, or live with the fact that samples will be broken until a new feature gets shipped in a release.

TL;DR: this should get fixed after a new release gets shipped.

**Comment by JustinBeckwith** [comment]
@ofrobots That shouldn't be the case.  The script we use to test samples does an `npm link ../` in the samples directory, pulling in the sources.

**Comment by ofrobots** [comment]
@JustinBeckwith can you point to the script? If what you say is true, then that script is no longer working correctly.

**Comment by ofrobots** [comment]
So here, an issue.. I am not sure if it is a problem yet. https://github.com/googleapis/nodejs-logging-bunyan/blob/202571b6aa3ea5c961dd55a37cf6a6ec1954644d/.kokoro/samples-test.sh#L38 This code installs a link. Then we run `npm run samples-tests` in the parent directory.

 https://github.com/googleapis/nodejs-logging-bunyan/blob/202571b6aa3ea5c961dd55a37cf6a6ec1954644d/.kokoro/samples-test.sh#L42

Which installs the link again. https://github.com/googleapis/nodejs-logging-bunyan/blob/202571b6aa3ea5c961dd55a37cf6a6ec1954644d/package.json#L45

**Comment by ofrobots** [comment]
The sample test seems to be passing on the [latest run]( https://source.cloud.google.com/results/invocations/de02fce5-f0bd-460c-a29e-95d7f5da4344/targets/cloud-devrel%2Fclient-libraries%2Fnodejs%2Fgoogleapis%2Fnodejs-logging-bunyan%2Fpresubmit%2Fnode8%2Fsamples-test/log) even though the status has been updated here.

**Comment by JustinBeckwith** [comment]
The system test failures appear to be unrelated, and I filed an issue in #204 to track that part.
